File: src/test/java/tests/api/PetStoreTest.java
package tests.api;

import baseTest.BaseTest;
import dataProviders.dataProviders.api.PetStoreDP;
import dataProviders.dataProvidersModels.api.PetStoreModel;
import io.restassured.response.Response;
import models.api.controllerPet.EndpointPet;
import models.api.controllerStore.EndpointStore;
import models.api.controllerUser.EndpointUser;
import models.api.pojo.POJOApiResponse;
import models.api.pojo.POJOOrder;
import models.api.pojo.POJOPet;
import models.api.pojo.POJOUser;
import org.testng.annotations.Test;
import utils.apiBodyComparer.ApiBodyComparer;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class PetStoreTest extends BaseTest {
    @Test(dataProvider = "sendGetRequestUserLoginDP", dataProviderClass = PetStoreDP.class)
    public void sendGetRequestUserLoginAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointUser().sendGetRequestUserLogin(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendPostRequestUserDP", dataProviderClass = PetStoreDP.class)
    public void sendPostRequestUserAndCheckResponse(PetStoreModel petStoreModel) {
        POJOApiResponse actualResponse = new EndpointUser().sendPostRequestUser(petStoreModel).getApiResponsePOJO();
        POJOApiResponse expectedResponse = new POJOApiResponse().deserialize(petStoreModel.getApiResponseBodyPath(), POJOApiResponse.class);
        assertThat(ApiBodyComparer.getPrettyLog(actualResponse)).isEqualTo(ApiBodyComparer.getPrettyLog(expectedResponse));
    }

    @Test(dataProvider = "sendGetRequestUserDP", dataProviderClass = PetStoreDP.class)
    public void sendGetRequestUserAndCheckResponse(PetStoreModel petStoreModel) {
        POJOUser actualResponse = new EndpointUser().sendGetRequestUser(petStoreModel).getUserPOJO();
        POJOUser expectedResponse = new POJOUser().deserialize(petStoreModel.getApiResponseBodyPath(), POJOUser.class);
        assertThat(ApiBodyComparer.getPrettyLog(actualResponse)).isEqualTo(ApiBodyComparer.getPrettyLog(expectedResponse));
    }

    @Test(dataProvider = "sendPutRequestUserDP", dataProviderClass = PetStoreDP.class)
    public void sendPutRequestUserAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointUser().sendPutRequestUser(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendDeleteRequestUserDP", dataProviderClass = PetStoreDP.class)
    public void sendDeleteRequestUserAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointUser().sendDeleteRequestUser(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendPostRequestPetDP", dataProviderClass = PetStoreDP.class)
    public void sendPostRequestPetAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointPet().sendPostRequestPet(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendGetRequestPetFindByStatusDP", dataProviderClass = PetStoreDP.class)
    public void sendGetRequestPetFindByStatusAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointPet().sendGetRequestPetFindByStatus(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendGetRequestPetFindByIdDP", dataProviderClass = PetStoreDP.class)
    public void sendGetRequestPetFindByIdAndCheckResponse(PetStoreModel petStoreModel) {
        POJOPet actualResponse = new EndpointPet().sendGetRequestPetFindById(petStoreModel).getPetPOJO();
        POJOPet expectedResponse = new POJOPet().deserialize(petStoreModel.getApiResponseBodyPath(), POJOPet.class);
        assertThat(ApiBodyComparer.getPrettyLog(actualResponse)).isEqualTo(ApiBodyComparer.getPrettyLog(expectedResponse));
    }

    @Test(dataProvider = "sendPutRequestPetDP", dataProviderClass = PetStoreDP.class)
    public void sendPutRequestPetAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointPet().sendPutRequestPet(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendDeleteRequestPetDP", dataProviderClass = PetStoreDP.class)
    public void sendDeleteRequestPetAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointPet().sendDeleteRequestPet(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test()
    public void sendGetRequestStoreInventoryAndCheckStatusCode() {
        Response actualResponse = new EndpointStore().sendGetRequestStoreInventory().getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendPostRequestStoreOrderDP", dataProviderClass = PetStoreDP.class)
    public void sendPostRequestStoreOrderAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointStore().sendPostRequestStoreOrder(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test(dataProvider = "sendGetRequestStoreOrderDP", dataProviderClass = PetStoreDP.class)
    public void sendGetRequestStoreOrderAndCheckResponse(PetStoreModel petStoreModel) {
        POJOOrder actualResponse = new EndpointStore().sendGetRequestStoreOrder(petStoreModel).getOrderPOJO();
        POJOOrder expectedResponse = new POJOOrder().deserialize(petStoreModel.getApiResponseBodyPath(), POJOOrder.class).replaceShipDate(actualResponse);
        assertThat(ApiBodyComparer.getPrettyLog(actualResponse)).isEqualTo(ApiBodyComparer.getPrettyLog(expectedResponse));
    }

    @Test(dataProvider = "sendDeleteRequestStoreOrderDP", dataProviderClass = PetStoreDP.class)
    public void sendDeleteRequestStoreOrderAndCheckStatusCode(PetStoreModel petStoreModel) {
        Response actualResponse = new EndpointStore().sendDeleteRequestStoreOrder(petStoreModel).getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }

    @Test()
    public void sendGetRequestUserLogoutAndCheckStatusCode() {
        Response actualResponse = new EndpointUser().sendGetRequestUserLogout().getResponse();
        assertThat(actualResponse.getStatusCode()).isEqualTo(200);
    }
}

File: ts-playwright-test-automation-framework/src/test/code/tests/api/PetStoreTest.ts
import { BaseTest } from '../../baseTest/BaseTest';
import { APIResponse } from "playwright"
import { EndpointPet } from '../../../../main/code/models/api/controllerPet/EndpointPet';
import { expect } from 'playwright/test';
import { POJOPet } from '../../../../main/code/models/api/pojo/POJOPet';
import { PetStoreModel } from '../../../../main/code/dataProviders/dataProvidersModels/api/PetStoreModel';
import { EndpointStore } from '../../../../main/code/models/api/controllerStore/EndpointStore';
import { EndpointUser } from '../../../../main/code/models/api/controllerUser/EndpointUser';
import { POJOOrder } from '../../../../main/code/models/api/pojo/POJOOrder';
import { POJOUser } from '../../../../main/code/models/api/pojo/POJOUser';
import { POJOApiResponse } from '../../../../main/code/models/api/pojo/POJOApiResponse';

export class PetStoreTest extends BaseTest {
    public async sendGetRequestUserLoginAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendGetRequestUserLoginAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointUser().sendGetRequestUserLogin(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendPostRequestUserAndCheckResponse(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendPostRequestUserAndCheckResponse');

        let actualResponse: POJOApiResponse = (await new EndpointUser().sendPostRequestUser(model)).getPOJOApiResponse();
        let expectedResponse: POJOApiResponse = POJOApiResponse.fromFile(model.apiResponseBodyPath!);
        expect(actualResponse).toMatchObject(JSON.parse(JSON.stringify(expectedResponse)));
    }

    public async sendGetRequestUserAndCheckResponse(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendGetRequestUserAndCheckResponse');

        let actualResponse: POJOUser = (await new EndpointUser().sendGetRequestUser(model)).getPOJOUser();
        let expectedResponse: POJOUser = POJOUser.fromFile(model.apiResponseBodyPath!);
        expect(actualResponse).toMatchObject(JSON.parse(JSON.stringify(expectedResponse)));
    }

    public async sendPutRequestUserAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendPutRequestUserAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointUser().sendPutRequestUser(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendDeleteRequestUserAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendDeleteRequestUserAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointUser().sendDeleteRequestUser(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendPostRequestPetAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendPostRequestPetAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointPet().sendPostRequestPet(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendGetRequestPetFindByStatusAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendGetRequestPetFindByStatusAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointPet().sendGetRequestPetFindByStatus(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendGetRequestPetFindByIdAndCheckResponse(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendGetRequestPetFindByIdAndCheckResponse');

        let actualResponse: POJOPet = (await new EndpointPet().sendGetRequestPetFindById(model)).getPOJOPet();
        let expectedResponse: POJOPet = POJOPet.fromFile(model.apiResponseBodyPath!);
        expect(actualResponse).toMatchObject(JSON.parse(JSON.stringify(expectedResponse)));
    }

    public async sendPutRequestPetAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendPutRequestPetAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointPet().sendPutRequestPet(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendDeleteRequestPetAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendDeleteRequestPetAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointPet().sendDeleteRequestPet(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendGetRequestStoreInventoryAndCheckStatusCode(): Promise<void> {
        let actualResponse: APIResponse = (await new EndpointStore().sendGetRequestStoreInventory()).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendPostRequestStoreOrderAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendPostRequestStoreOrderAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointStore().sendPostRequestStoreOrder(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendGetRequestStoreOrderAndCheckResponse(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendGetRequestStoreOrderAndCheckResponse');

        let actualResponse: POJOOrder = (await new EndpointStore().sendGetRequestStoreOrder(model)).getPOJOOrder();
        let expectedResponse: POJOOrder = POJOOrder.fromFile(model.apiResponseBodyPath!);
        expect(actualResponse).toMatchObject(JSON.parse(JSON.stringify(expectedResponse)));
    }

    public async sendDeleteRequestStoreOrderAndCheckStatusCode(): Promise<void> {
        const model = this.getModel<PetStoreModel>('sendDeleteRequestStoreOrderAndCheckStatusCode');

        let actualResponse: APIResponse = (await new EndpointStore().sendDeleteRequestStoreOrder(model)).getResponse();
        expect(actualResponse.status()).toBe(200)
    }

    public async sendGetRequestUserLogoutAndCheckStatusCode(): Promise<void> {
        let actualResponse: APIResponse = (await new EndpointUser().sendGetRequestUserLogout()).getResponse();
        expect(actualResponse.status()).toBe(200)
    }
}